File: app/models/landing_page_builder/color.rb
module LandingPageBuilder
    class Color
        include Virtus.model

        attribute :red, Integer, default: 255
        attribute :green, Integer, default: 255
        attribute :blue, Integer, default: 255
        attribute :alpha, Float, default: 1.0

        def ==(other)
            (
                self.red == other.red &&
                self.green == other.green &&
                self.blue == other.blue &&
                self.alpha == other.alpha
            )
        end

    end
end

File: graphql-gateway/src/experiences/models/Color.js
import {
	GraphQLInt,
	GraphQLFloat,
	GraphQLNonNull,
	GraphQLObjectType
} from 'graphql'

const Color = new GraphQLObjectType({
    name: 'Color',
    fields: () => ({
        red: {
            type: new GraphQLNonNull(GraphQLInt),
            resolve: data => data['red']
        },
        green: {
            type: new GraphQLNonNull(GraphQLInt),
            resolve: data => data['green']
        },
        blue: {
            type: new GraphQLNonNull(GraphQLInt),
            resolve: data => data['blue']
        },
        alpha: {
            type: new GraphQLNonNull(GraphQLFloat),
            resolve: data => data['alpha']
        }
    })
})

export default Color
